lang/java/avro/src/main/java/org/apache/avro/io/JsonEncoder.java

```diff
@@ -208,7 +208,7 @@ public void writeLong(long n) throws IOException {
   @Override
   public void writeFloat(float f) throws IOException {
     parser.advance(Symbol.FLOAT);
-    out.writeNumber(f);
+    out.writeNumber(f + 0d);
```

Comment:
Won't this add more digits / convert a float to a double?

I though that's what we're trying to prevent.

Reply:
We want to prevent the fact that in Java, `Float.toString(33.33000183105469f)` return "33.33" instead of the whole number.
So, the cast to Double trigger Double.toString method instead which keep precision, and write 33.33000183105469.
The "parser.advance(Symbol.FLOAT)" ensure we won't cast it to double at reading.